### ISSUE TITLE
[openweathermap] Binding interval with new API 3.0 is not working

### ISSUE THREAD

**Issue by erikschitterer** [openhab/openhab-addons#17092]
If I use the binding with the new 3.0 API call, the default interval 10 min are not working. If I switch back to version 2.5, the interval is working fine.

## Your Environment
Openhab 4.3.0 Snapshot

**Comment by skydiver66** [comment]
same like me...

When I reactivate the binding, weather will be updated, but only once.

**Comment by lsiepel** [comment]
Could you provide a debug log?

**Comment by erikschitterer** [comment]
Hi @lsiepel, I run the binding in debug mode, baut no errors in the log.

Workarround: I create a rule to disable/enable the api thing and create a schedule to start this rule every 15 minutes.

Script:

var thingMgr = osgi.getService('org.openhab.core.thing.ThingManager');
var ThingUID = Java.type('org.openhab.core.thing.ThingUID');
thingMgr.setEnabled(new ThingUID('openweathermap:weather-api:8de88b7805'), false);
setTimeout(thingMgr.setEnabled(new ThingUID('openweathermap:weather-api:8de88b7805'), true),10000);

Rule/Cron: 0 0/15 * * * ? *

**Comment by lsiepel** [comment]
Ah, a `NullPointerException` Unfortunately this binding lacks null annotations, so it is not easy to fix without specific binding knowledge.  If you put the binding in trace logging it should show the response from OWM and that should give me enough to try to fix this. 
Already started to do some cleanup to this binding, so hopefully with your log i can give you a patched jar by tomorrow.

**Comment by erikschitterer** [comment]
Hi @lsiepel , here the trace-log.

[OWM-trace.log](https://github.com/user-attachments/files/16436789/OWM-trace.log)

Thanks a lot!

**Comment by lsiepel** [comment]
Weird, there are atleast two issues. 
1. The returned json has a `data` field and a `current` is expected. The rest of the schema seems correct
2. When the returned json  is cor5rected it gives an `JsonParseException` due to the array.

I wrote some tests and have found more issues, but they are not related. Need some more time to discover why this happens.

**Comment by erikschitterer** [comment]
no problem, I use the workaround and use the rule/script for interval.

**Comment by lsiepel** [comment]
Would be nice if you can confirm this issue is fixed iwth this jar: https://1drv.ms/u/s!AnMcxmvEeupwjvw-BPBJwbpfUh-pWQ?e=UaqTdj

- Uninstall your current openweathermap binding
- Drop the jar file in your addon folder.

Please report back any issues (with trace log)

**Comment by erikschitterer** [comment]
I have download the file and follow the instructions. The installed and used version is now 4.3.0.202407312139

After restart I got the still some NullPointerException errors

Interval still not working
[OWM-trace.log](https://github.com/user-attachments/files/16451424/OWM-trace.log)

**Comment by lsiepel** [comment]
Ah, that is similar but new issue. The 3.0 api does not support hourly channels. I’ll add a check.

**Comment by skydiver66** [comment]
Hi, as I remember, the forecast jason response body from 3.0 api is differnt to the 2.5 api.

**Comment by lsiepel** [comment]
Just created a new verison, You can use the same link for the jar file to test. 

Would be nice if you can first test to see if it works as is. It should show warnings in your log that the hourly channels are not supported by v3 api.
When confirmed, could you remove the hourly channels to make shure none of them are linekd and observe that the log no longer shows the before mentioned warnings?

**Comment by skydiver66** [comment]
the weather data updated again, but I received this now (for all hours): 

2024-08-01 12:56:34.619 [WARN ] [.OpenWeatherMapOneCallHistoryHandler] - No weather data available for channel openweathermap:onecall-history:254af30069:local:historyHours03#temperature, possible cause: api v3.0 does not support it

2024-08-01 12:56:34.621 [WARN ] [.OpenWeatherMapOneCallHistoryHandler] - No weather data available for channel openweathermap:onecall-history:254af30069:local:historyHours03#humidity, possible cause: api v3.0 does not support it

**Comment by erikschitterer** [comment]
ok, here my results:

1.) the interval is now wirking!!!!!!!

2.) Test the new jar file with the history_hoursXX items: 
[OWM-Trace-Step1-with-hours.txt](https://github.com/user-attachments/files/16455193/OWM-Trace-Step1-with-hours.txt)

3.) Delete all history_hoursXX items **AND** change the dimension for all forecast_hourXX_visibility items from "m" to "%":
[OWM-Trace-Step2-without-hours.txt](https://github.com/user-attachments/files/16455212/OWM-Trace-Step2-without-hours.txt)

Now, I see no errors or warnings. For me, it seems, that the issue is solved

**Comment by lsiepel** [comment]
> the weather data updated again, but I received this now (for all hours):
> 
> 2024-08-01 12:56:34.619 [WARN ] [.OpenWeatherMapOneCallHistoryHandler] - No weather data available for channel openweathermap:onecall-history:254af30069:local:historyHours03#temperature, possible cause: api v3.0 does not support it
> 
> 2024-08-01 12:56:34.621 [WARN ] [.OpenWeatherMapOneCallHistoryHandler] - No weather data available for channel openweathermap:onecall-history:254af30069:local:historyHours03#humidity, possible cause: api v3.0 does not support it

Yes that is expected. You should remove those hourly items/links as they are no longer part of the v3 api

**Comment by lsiepel** [comment]
> ok, here my results:
> 
> 1.) the interval is now wirking!!!!!!!
> 
> 2.) Test the new jar file with the history_hoursXX items: [OWM-Trace-Step1-with-hours.txt](https://github.com/user-attachments/files/16455193/OWM-Trace-Step1-with-hours.txt)
> 
> 3.) Delete all history_hoursXX items **AND** change the dimension for all forecast_hourXX_visibility items from "m" to "%": [OWM-Trace-Step2-without-hours.txt](https://github.com/user-attachments/files/16455212/OWM-Trace-Step2-without-hours.txt)
> 
> Now, I see no errors or warnings. For me, it seems, that the issue is solved

Nothing was changed to the dimension. Both api versions use the same dimension, so that was probably already something wrong in your setup?

**Comment by erikschitterer** [comment]
> > ok, here my results:
> > 1.) the interval is now wirking!!!!!!!
> > 2.) Test the new jar file with the history_hoursXX items: [OWM-Trace-Step1-with-hours.txt](https://github.com/user-attachments/files/16455193/OWM-Trace-Step1-with-hours.txt)
> > 3.) Delete all history_hoursXX items **AND** change the dimension for all forecast_hourXX_visibility items from "m" to "%": [OWM-Trace-Step2-without-hours.txt](https://github.com/user-attachments/files/16455212/OWM-Trace-Step2-without-hours.txt)
> > Now, I see no errors or warnings. For me, it seems, that the issue is solved
> 
> Nothing was changed to the dimension. Both api versions use the same dimension, so that was probably already something wrong in your setup?

You can see in the first log, that there was errors with "_visibility" items and the dimensions. 

> 2024-08-01 09:47:35.935 [WARN ] [penhab.core.library.items.NumberItem] - Failed to update item 'One_Call_API_weather_and_forecast_ForecastHours01_Visibility' because '100 %' could not be converted to the item unit 'm'

Currently, the forecast_hoursXX_visibility is coming back from the API with the dimension "%" and not "m".
![image](https://github.com/user-attachments/assets/9fdcd9f2-2758-415c-9a83-a8fdb723dcdc)


In the API description I see, that the dimension must be "m" : (source: https://openweathermap.org/api/one-call-3)
![image](https://github.com/user-attachments/assets/9128db3c-3ab4-4ca2-8f80-f557f05ae135)

**Comment by skydiver66** [comment]
Die Sichtweite sollte in Meter ausgegeben werden, vielleicht ist dein Item falsch definiert.
Ich hab das so und es läuft:
Number:Length CurrentVisibility "Sichtweite [%.0f km]" <zoom> (Weather){channel="openweathermap:onecall:254axxx:local:current#visibility"}

**Comment by lsiepel** [comment]
> > > ok, here my results:
> > > 1.) the interval is now wirking!!!!!!!
> > > 2.) Test the new jar file with the history_hoursXX items: [OWM-Trace-Step1-with-hours.txt](https://github.com/user-attachments/files/16455193/OWM-Trace-Step1-with-hours.txt)
> > > 3.) Delete all history_hoursXX items **AND** change the dimension for all forecast_hourXX_visibility items from "m" to "%": [OWM-Trace-Step2-without-hours.txt](https://github.com/user-attachments/files/16455212/OWM-Trace-Step2-without-hours.txt)
> > > Now, I see no errors or warnings. For me, it seems, that the issue is solved
> > 
> > 
> > Nothing was changed to the dimension. Both api versions use the same dimension, so that was probably already something wrong in your setup?
> 
> You can see in the first log, that there was errors with "_visibility" items and the dimensions.
> 
> > 2024-08-01 09:47:35.935 [WARN ] [penhab.core.library.items.NumberItem] - Failed to update item 'One_Call_API_weather_and_forecast_ForecastHours01_Visibility' because '100 %' could not be converted to the item unit 'm'
> 
> Currently, the forecast_hoursXX_visibility is coming back from the API with the dimension "%" and not "m". ![image](https://private-user-images.githubusercontent.com/57366719/354208817-9fdcd9f2-2758-415c-9a83-a8fdb723dcdc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1MTQ2OTAsIm5iZiI6MTcyMjUxNDM5MCwicGF0aCI6Ii81NzM2NjcxOS8zNTQyMDg4MTctOWZkY2Q5ZjItMjc1OC00MTVjLTlhODMtYThmZGI3MjNkY2RjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAxVDEyMTMxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxNDY0OWI4MGJjYjI1ZWNlNjIyNThiNmI1OTY4ZDgwODVkOTFhZjQ1NGFjNzUzYWE4ZWJiZDM4YTc1ZDVlODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GceTlRas70bm3IhTTf0p6H_hFLXCAtYHUt_J2DqVBsg)
> 
> In the API description I see, that the dimension must be "m" : (source: https://openweathermap.org/api/one-call-3) ![image](https://private-user-images.githubusercontent.com/57366719/354208492-9128db3c-3ab4-4ca2-8f80-f557f05ae135.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1MTQ2OTAsIm5iZiI6MTcyMjUxNDM5MCwicGF0aCI6Ii81NzM2NjcxOS8zNTQyMDg0OTItOTEyOGRiM2MtM2FiNC00Y2EyLThmODAtZjU1N2YwNWFlMTM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAxVDEyMTMxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0ZGUxMWEyMTViNjYxNzgzMzE4ZTQzODkzMjRiZjY1ZmI1NDRmOWRjNTZjYWU0ZDRkZmNlMzg2OTAzZjc4NmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4MkjdNxcoGudpWtpAraP4vxjfV5cpNWMUp3ATc1K8mE)

Forecast is a different endpoint not related. I might have a quick peek if able to fix that too

**Comment by lsiepel** [comment]
> Die Sichtweite sollte in Meter ausgegeben werden, vielleicht ist dein Item falsch definiert. Ich hab das so und es läuft: Number:Length CurrentVisibility "Sichtweite [%.0f km]" (Weather){channel="openweathermap:onecall:254axxx:local:current#visibility"}

English please

**Comment by erikschitterer** [comment]
> Die Sichtweite sollte in Meter ausgegeben werden, vielleicht ist dein Item falsch definiert. Ich hab das so und es läuft: Number:Length CurrentVisibility "Sichtweite [%.0f km]" (Weather){channel="openweathermap:onecall:254axxx:local:current#visibility"}

I have no problem with the items current_visibility and history_visibility. Both have the dimension "m" and I use the same pattern as you. My problem is with the forecast_hoursXX_visibilty items. I have check it again, if I set the item to dimension "m" the an error appears in the log via the update

**Comment by lsiepel** [comment]
The error you receive:
``` 
2024-08-01 09:47:38.949 [WARN ] [penhab.core.library.items.NumberItem] - Failed to update item 'One_Call_API_weather_and_forecast_ForecastHours45_Visibility' because '43 %' could not be converted to the item unit 'm'
```

The AP returns `meters`:
https://openweathermap.org/api/one-call-3#example

The DTO has an `int` to store the value, so `no decimal` for percentage:
https://github.com/openhab/openhab-addons/blob/87eb3a5feb4051789c36e4a1b6dbe69ffa3e1459/bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/dto/onecall/Hourly.java#L34

The channel is defined here and correctly shows `length` and not percentage.
https://github.com/openhab/openhab-addons/blob/87eb3a5feb4051789c36e4a1b6dbe69ffa3e1459/bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/thing/channel-types.xml#L865C2-L870C2

The value to that channel is set here:
https://github.com/openhab/openhab-addons/blob/87eb3a5feb4051789c36e4a1b6dbe69ffa3e1459/bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapWeatherAndForecastHandler.java#L310-L316

I must say that i cannot find the issue here with the error you have,. Can you somehow get a debug log and show your item declaration?

**Comment by mherwege** [comment]
In the screenshots, I see the items being defined as Number:Dimensionless and not Number:Length. m is not a Dimensionless unit.

**Comment by erikschitterer** [comment]
Hi, I changed one item back to Number:Length with unit "m":

Channel config:
![image](https://github.com/user-attachments/assets/7f84dd32-01b1-49ad-bb28-61d20fcf3d35)

Channel Profile
![image](https://github.com/user-attachments/assets/8cc52299-c88d-496d-9f94-a4cc060db385)


Item Config:
![image](https://github.com/user-attachments/assets/7d37ca86-db0e-45b2-a00b-c9d9e3d1b5cf)

DEBUG Error message:

```
2024-08-02 17:03:50.922 [DEBUG] [handler.OpenWeatherMapOneCallHandler] - OneCallHandler: updateChannel openweathermap:onecall:9ad38318c2:local:forecastHours01#visibility, groupID forecastHours01
2024-08-02 17:03:50.922 [DEBUG] [handler.OpenWeatherMapOneCallHandler] - Update channel 'visibility' of group 'forecastHours01' with new state '0 %'.
2024-08-02 17:03:50.923 [WARN ] [penhab.core.library.items.NumberItem] - Failed to update item 'One_Call_API_weather_and_forecast_ForecastHours01_Visibility' because '0 %' could not be converted to the item unit 'm'


```

You see, the channel update send the state **'0 %'**. Who or what set this dimension?????

**Comment by mherwege** [comment]
You changed it for a group item, that gets calculated based on underlying item values. If these are %, you will not get m.

**Comment by erikschitterer** [comment]
I don't understand your analyze with a group item. I changed back the single item `"One_Call_API_weather_and_forecast_ForecastHours01_Visibility"` to Number:Lenght. This Item has a link to the channel "`openweathermap:onecall:9ad38318c2:local:forecastHours01#visibility (Number:Length)"` from the "`One Call API weather and forecast"` thing. This configuraten works fine the last 5 years with the 2.5 binding. After the change to 3.0 and the fix of the interval and the deletion of all `"history_hoursXX" `items, the `"forecast_hoursXX_visibility"` warning with the dimension-problem appear.

I have no major problem with the hourly visibility in "%", because I use only the general history_visibility & current_visibility items for visualizion. And these are still in "m"

**Comment by mherwege** [comment]
> I don't understand your analyze with a group item. I changed back the single item `"One_Call_API_weather_and_forecast_ForecastHours01_Visibility"` to Number:Lenght. This Item has a link to the channel "`openweathermap:onecall:9ad38318c2:local:forecastHours01#visibility (Number:Length)"` from the "`One Call API weather and forecast"` thing. This configuraten works fine the last 5 years with the 2.5 binding. After the change to 3.0 and the fix of the interval and the deletion of all `"history_hoursXX" `items, the `"forecast_hoursXX_visibility"` warning with the dimension-problem appear.
> 
> I have no major problem with the hourly visibility in "%", because I use only the general history_visibility & current_visibility items for visualizion. And these are still in "m"

My mistake with the group item. I deleted it straight after writing, but it looks like you had seen it. I was mislead by the log and the mention of group in it, but that is a channel group, not item group.

**Comment by erikschitterer** [comment]
no problem @mherwege 

btw: I have delete all `forecastHoursXX_visibility` items and rebuild them with the default dimensions were inherited from the channel (Number:Lenght). The warning is still the same.

Then I have created a complete new thing with new channels, the warning still available
